[http://sources.redhat.com/ml/crossgcc/2005-10/msg00035.html]

Message-ID: <434576E1.6020305@sscd.de>
Date: Thu, 06 Oct 2005 21:11:29 +0200
From: Alexander Sieb <sieb@sscd.de>
To: crossgcc@sourceware.org
Subject: Crosstool sh4-linux-gcc-4.0.2-glibc-2.3.5 patches

Hi,

attached you find the files I needed to add to crosstool-0.38
in order to build a sh4-linux gcc-4.0.2-glibc-2.3.5 tool chain.

Files attached:

sh4-gcc4.dat:
        * gcc4 needs --with-multilib-list=m4,m4-nofpu otherwise a linux
        kernel won't build as it uses the -m4-nofpu option.

gcc-pr21623.patch:
        * Kaz Kojima provided a patch [really, a workaround -dank] for http://gcc.gnu.org/PR21623

glibc-2.3.5-sh-memset.patch:
        * A patch for glibc-2.3.5 which corrects memset. From SUGIOKA Toshinobu.
        See content for rationale.

-- snip --

-- Here's patch 2 of 2, plus URLs showing where it is in CVS --

[http://sources.redhat.com/ml/glibc-cvs/2005-q3/msg00319.html]

Date: 24 Jul 2005 22:54:32 -0000
Message-ID: <20050724225432.2111.qmail@sourceware.org>
From: roland at sourceware dot org
To: glibc-cvs at sources dot redhat dot com
Subject: libc/sysdeps/sh memset.S

CVSROOT:	/cvs/glibc
Module name:	libc
Changes by:	roland@sources.redhat.com	2005-07-24 22:54:32

Modified files:
	sysdeps/sh     : memset.S 

Log message:
	2005-07-24  SUGIOKA Toshinobu  <sugioka@itonet.co.jp>
	
	* sysdeps/sh/memset.S (memset): Correct 2nd argument handling.

[http://sources.redhat.com/cgi-bin/cvsweb.cgi/libc/sysdeps/sh/memset.S.diff?cvsroot=glibc&r1=1.4&r2=1.5]

RCS file: /cvs/glibc/libc/sysdeps/sh/memset.S,v
retrieving revision 1.4
retrieving revision 1.5
diff -u -r1.4 -r1.5
--- libc/sysdeps/sh/memset.S	2003/04/29 22:47:18	1.4
+++ libc/sysdeps/sh/memset.S	2005/07/24 22:54:32	1.5
@@ -28,6 +28,7 @@
 	bt.s	L_byte_loop_init
 	mov	r4,r7
 
+	extu.b	r5,r5
 	swap.b	r5,r1
 	or	r1,r5
 	swap.w	r5,r1


[See also http://sources.redhat.com/ml/libc-alpha/2005-07/msg00051.html]
